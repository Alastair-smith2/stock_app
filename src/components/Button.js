import React, { Component } from "react";
import { View, Text, TouchableHighlight, StyleSheet } from "react-native";

export default class Button extends Component {
  constructor(props) {
    super(props);

    this.onPress = this.onPress.bind(this);
  }
  onPress = () => {
    this.props.buttonPress();
  };

  render() {
    return (
      <TouchableHighlight
        onPress={this.onPress}
        style={[styles.container, this.props.buttonStyle]}
      >
        <Text style={styles.label}>{this.props.title}</Text>
      </TouchableHighlight>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  label: {
    fontSize: 20,
    color: "#fff"
  }
});
