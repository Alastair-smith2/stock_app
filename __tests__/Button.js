import React from "react";
import { shallow } from "enzyme";

import Button from "../src/components/Button.js";

describe("The button component", () => {
  let wrapper;
  let onPressInput = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <Button title="Test Button" buttonPress={onPressInput} />
    );
    onPressInput.mockReset();
  });
  describe("It should render", () => {
    it("Should render", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("Features of the button", () => {
    it("Should have 1 touchable highlight", () => {
      expect(wrapper.find("TouchableHighlight")).toHaveLength(1);
    });

    it("Should an on press function that calls the parent function", () => {
      wrapper.find("TouchableHighlight").simulate("press");
      expect(onPressInput.mock.calls.length).toEqual(1);
    });

    it("Should have a label field that is inherited from props", () => {
      expect(wrapper.find("Text").contains("Test Button"));
    });
  });
});
