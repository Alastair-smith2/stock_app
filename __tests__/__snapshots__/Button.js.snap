// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The button component It should render Should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    buttonPress={[MockFunction]}
    title="Test Button"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.85,
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontSize": 20,
          }
        }
      >
        Test Button
      </Text>,
      "delayPressOut": 100,
      "onPress": [Function],
      "style": Array [
        Object {
          "flex": 1,
        },
        undefined,
      ],
      "underlayColor": "black",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "Test Button",
        "ellipsizeMode": "tail",
        "style": Object {
          "color": "#fff",
          "fontSize": 20,
        },
      },
      "ref": null,
      "rendered": "Test Button",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.85,
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontSize": 20,
            }
          }
        >
          Test Button
        </Text>,
        "delayPressOut": 100,
        "onPress": [Function],
        "style": Array [
          Object {
            "flex": 1,
          },
          undefined,
        ],
        "underlayColor": "black",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Test Button",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#fff",
            "fontSize": 20,
          },
        },
        "ref": null,
        "rendered": "Test Button",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
